### DISCUSSION TITLE
svg sprites works with turbo but not with webpack

### DISCUSSION THREAD

**Discussion by abaracedo** [vercel/next.js#53158] [Help] [answered]
### Summary

I'm using sprites SVG for improving performance. Using turbo I didn't need to change anything from the configuration and the svg are rendered fine.

This is my Icon component
```jsx
// src/components/Icon
import icons from './icons.svg';

export default function Icon({ type }) {
  return (
    <svg>
      <use href={`${icons.src}#${type}`} />
    </svg>
  );
}
```
Developing with Webpack I thrown an error about svg not being a valid type. I installed `@svgr/webpack` and I configured it on `next.config.js` file in this way.

```
webpack: (config) => {
    // Grab the existing rule that handles SVG imports
    const fileLoaderRule = config.module.rules.find((rule) => rule.test?.test?.('.svg'));

    config.module.rules.push(
      // Convert *.svg imports to React components
      {
        test: /\.svg$/i,
        issuer: fileLoaderRule.issuer,
        resourceQuery: { not: [...fileLoaderRule.resourceQuery.not, /url/] },
        loader: '@svgr/webpack',
      },
    );

    // Modify the file loader rule to ignore *.svg, since we have it handled now.
    fileLoaderRule.exclude = /\.svg$/i;

    return config;
  },
```

With that configuration for WebPack the error is gone but the svg file does not loads in the browser. The browser tries to fetch it from the public folder but the file does not exist in it as it exists on `src/components/Icon`.

I don't get it how it works with turbo but inspecting the code the application returns I see that the svg file comes from `_next/static/`.

So the thing is, is there a way I can load the file from `_next/static/` when using WebPack as the bundler?

**Comment by abaracedo** [comment, marked as answer]
After investigating I found that the issue can be solved for WebPack using asset module instead of the package `@svgr/webpack`
```
  webpack: (config) => {
    const fileLoaderRule = config.module.rules.find((rule) => rule.test?.test?.('.svg'));

    config.module.rules.push({
      test: /\.svg/i,
      type: 'asset',
    });

    fileLoaderRule.exclude = /\.svg$/i;

    return config;
  },
```

**Reply by Karina1703** [reply]
Please, can you share your full config? I'm trying to add sprite svg to my next app, but svgr doesn't want to work with sprites

**Reply by abaracedo** [reply]
Sure, but I'm not using `@svgr/webkapck` anymore.

```
const nextConfig = {
  reactStrictMode: true,
  output: 'standalone',
  webpack: (config) => {
    const fileLoaderRule = config.module.rules.find((rule) => rule.test?.test?.('.svg'));

    config.module.rules.push({
      test: /\.svg/i,
      type: 'asset',
    });

    fileLoaderRule.exclude = /\.svg$/i;

    return config;
  },
  eslint: {
    ignoreDuringBuilds: true,
  },
};
```
